### PR TITLE
build: car details

### PR DESCRIPTION
**Pull request by abdou059**
Bring all car data

### CODE REVIEW

File: server/src/controllers/cars/gitCarsId.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Request, Response, NextFunction } from 'express';
+import { getCarById } from '../../services';
+
+const getCarsId = async (req: Request, res: Response, next: NextFunction) => {
+  const { id } = req.params;
+
+  const reus = await getCarById(id);
```
Comment (mai-mohammed) [inline server/src/controllers/cars/gitCarsId.ts]: check the result if its empty(not found) or not

**Review by husamkamal — APPROVED** [review]

**Review by tariqabusada — APPROVED** [review]

File: server/src/services/cars/getCar.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { Car } from '../../db/models';
+
+const getCarById = async (id) => {
+  const carId = await Car.findByPk(id);
```
Comment (mai-mohammed) [inline server/src/services/cars/getCar.ts]: rename this to result or car

Comment (LinaYahya) [reply server/src/services/cars/getCar.ts]: also include image model to get image for this car

File: server/src/routes/cars.ts [hunk 3]
```diff
@@ -1,9 +1,10 @@
 import { Router } from 'express';
 import ExpressWrapper from './ExpressWrapper';
-import { getFilteredCars } from '../controllers';
+import { getFilteredCars, getCarsId } from '../controllers';
 
 const carsRouter = Router();
 
 carsRouter.get('/cars', ExpressWrapper(getFilteredCars));
+carsRouter.get('/cars/:id', ExpressWrapper(getCarsId));
```
Comment (LinaYahya) [inline server/src/routes/cars.ts]:
```suggestion
carsRouter.get('/cars/:id', ExpressWrapper(getCarsById));
```

File: server/src/controllers/cars/index.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
 import getFilteredCars from './getFilteredCars';
+import getCarsId from './gitCarsId';
```
Comment (LinaYahya) [inline server/src/controllers/cars/index.ts]:
```suggestion
import getCarsId from './getCarsId';
```

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by mai-mohammed — APPROVED** [review]

File: server/__tests__/carsdetails.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { describe, expect, test } from '@jest/globals';
```
Comment (LinaYahya) [inline server/__tests__/carsdetails.ts]: beforeAll and afterAll required

File: server/src/controllers/cars/getCarsById.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { Request } from 'express';
+import { getCarInfo } from '../../services';
+
+const getCarsById = async (req: Request) => {
+  const { id } = req.params;
```
Comment (LinaYahya) [inline server/src/controllers/cars/getCarsById.ts:6]: try to validate `id` to ensure it's a number, If not end response as a bad request

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by LinaYahya — APPROVED** [review]

**Pull request merged**